src/AWS.Deploy.Orchestration/OptionSettingHandler.cs

```diff
@@ -333,5 +349,85 @@ public bool IsSummaryDisplayable(Recommendation recommendation, OptionSettingIte
+            {
+                return false;
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.List))
+            {
+                var currentSet = GetOptionSettingValue<SortedSet<string>>(recommendation, optionSetting);
+                var defaultSet = GetOptionSettingDefaultValue<SortedSet<string>>(recommendation, optionSetting);
+
+                if (defaultSet == null)
+                    return currentSet?.Count > 0;
+
+                // return true if both have different lengths or all elements in currentSet are not present in defaultSet
+                return defaultSet.Count != currentSet?.Count || !currentSet.All(x => defaultSet.Contains(x));
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.KeyValue))
+            {
+                var currentDict = GetOptionSettingValue<Dictionary<string, string>>(recommendation, optionSetting);
+                var defaultDict = GetOptionSettingDefaultValue<Dictionary<string, string>>(recommendation, optionSetting);
+
+                if (defaultDict == null)
+                    return currentDict?.Count > 0;
+
+                // return true if both have different lengths or all keyValue pairs are not equal between currentDict and defaultDict
+                return defaultDict.Count != currentDict?.Count ||
+                    !currentDict.All(keyPair => defaultDict.ContainsKey(keyPair.Key) && string.Equals(defaultDict[keyPair.Key], currentDict[keyPair.Key]));
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.Int))
+            {
+                var currentValue = GetOptionSettingValue<int>(recommendation, optionSetting);
+                var defaultValue = GetOptionSettingDefaultValue<int>(recommendation, optionSetting);
+                return defaultValue != currentValue;
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.Double))
+            {
+                var currentValue = GetOptionSettingValue<double>(recommendation, optionSetting);
+                var defaultValue = GetOptionSettingDefaultValue<double>(recommendation, optionSetting);
+                return defaultValue != currentValue;
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.Bool))
+            {
+                var currentValue = GetOptionSettingValue<bool>(recommendation, optionSetting);
+                var defaultValue = GetOptionSettingDefaultValue<bool>(recommendation, optionSetting);
+                return defaultValue != currentValue;
+            }
+
+            if (optionSetting.Type.Equals(OptionSettingValueType.String))
+            {
+                var currentValue = GetOptionSettingValue<string>(recommendation, optionSetting);
+                var defaultValue = GetOptionSettingDefaultValue<string>(recommendation, optionSetting);
+
+                if (string.IsNullOrEmpty(currentValue) && string.IsNullOrEmpty(defaultValue))
```

Comment:
I don't think we should treat null or empty then same. An option setting could be set to empty but doesn't have a default value. That could be an updated option setting.

Reply:
Can you point to an example where the default value for a string option setting is null, but it is still relevant to the deployment when it holds a `string.empty` value?


For example, consider the [StartCommand](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppAppRunner.recipe#L133) and [EncryptionKmsKey](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppAppRunner.recipe#L282) in the AppRunner recipe. They do not have an explicit default value (i.e the default values are `null`). However, the CDK template will only use these settings if they are not null or empty (effectively treating null and empty as same) [proof#1](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppAppRunner/Generated/Recipe.cs#L184), [proof#2](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppAppRunner/Generated/Recipe.cs#L203)